transport/src/main/java/io/netty/channel/EventLoopGroup.java

```diff
@@ -49,4 +49,10 @@
      */
     @Deprecated
     ChannelFuture register(Channel channel, ChannelPromise promise);
+
+    /**
+     * Return the defaults {@link ChannelFactory} for channels to be created in the context of this group
+     * @return a default channel factory or null if not provided
+     */
+    ChannelFactory defaultChannelFactory(boolean serverSide);
```

Comment:
@normanmaurer that you for your review

I understand this point.
Netty is compiled for Java6 (I think) and so defaults method are not supported.
Another way to implement this feature would be to create a global registry which maps EventLoopGroup class with the Channel class.
This registry will be filled in static {} blocks in EventLoopGroup subclasses

I can file a new PR in order to demostrate this alternative approach